kadi/commands/core.py

```diff
@@ -0,0 +1,824 @@
+        new_col = np.concatenate([t[name] for t in tables])
+        new_cols.append(new_col)
+
+    out = table0.__class__(new_cols, names=names0)
+    return out
+
+
+def read_backstop(backstop):
+    """Read ``backstop`` and return a ``CommandTable``.
+
+    The ``backstop`` argument can be either be a string file name or a backstop
+    table from ``parse_cm.read_backstop``.
+
+    This function is a wrapper around ``get_cmds_from_backstop`` but follows a
+    more typical naming convention.
+
+    :param backstop: str or Table
+    :returns: :class:`~kadi.commands.commands.CommandTable` of commands
+    """
+    return get_cmds_from_backstop(backstop)
+
+
+def get_cmds_from_backstop(backstop, remove_starcat=False):
+    """
+    Initialize a ``CommandTable`` from ``backstop``, which can either
+    be a string file name or a backstop table from ``parse_cm.read_backstop``.
+
+    :param backstop: str or Table
+    :param remove_starcat: remove star catalog command parameters (default=False)
+    :returns: :class:`~kadi.commands.commands.CommandTable` of commands
+    """
+    if isinstance(backstop, Path):
+        backstop = str(backstop)
+
+    if isinstance(backstop, (str, list)):
+        from parse_cm import read_backstop
+        bs = read_backstop(backstop)
+    elif isinstance(backstop, Table):
+        bs = backstop
+    else:
+        raise ValueError(f'`backstop` arg must be a string filename or '
+                         f'a backstop Table')
+
+    n_bs = len(bs)
+    out = {}
+    # Set idx to -1 so it does not match any real idx
+    out['idx'] = np.full(n_bs, fill_value=-1, dtype=np.int32)
+    out['date'] = np.chararray.encode(bs['date'])
+    out['type'] = np.chararray.encode(bs['type'])
+    out['tlmsid'] = np.chararray.encode(bs['tlmsid'])
+    out['scs'] = bs['scs'].astype(np.uint8)
+    out['step'] = bs['step'].astype(np.uint16)
+    out['time'] = CxoTime(bs['date'], format='date').secs
+    # Set timeline_id to 0, does not match any real timeline id
+    out['timeline_id'] = np.zeros(n_bs, dtype=np.uint32)
+    out['vcdu'] = bs['vcdu'].astype(np.int32)
+    out['params'] = bs['params']
+
+    if remove_starcat:
+        # Remove the lengthy parameters in star catalog but leave the command
+        for idx in np.flatnonzero(bs['type'] == 'MP_STARCAT'):
+            out['params'][idx] = {}
+
+    # Backstop has redundant param keys, get rid of them here
+    for params in out['params']:
+        for key in ('tlmsid', 'step', 'scs'):
+            params.pop(key, None)
+
+        # Match the hack in update_commands which swaps in "event_type" for "type"
+        # for orbit event commands
+        if 'type' in params:
+            params['event_type'] = params['type']
+            del params['type']
+
+    out = CommandTable(out)
+    out['time'].info.format = '.3f'
+
+    return out
+
+
+def _find(start=None, stop=None, inclusive_stop=False, idx_cmds=None,
+          pars_dict=None, **kwargs):
```

Comment:
Might need some docstring help with idx_cmds and pars_dict.

Reply:
Done.